test/test_sparse.py

```diff
@@ -3481,6 +3481,25 @@ def test_softmax_zero_nnz(self, device, dtype):
         out = torch.sparse.softmax(t, 0)
         self.assertEqual(out.to_dense(), torch.zeros_like(t))
 
+        t = t.requires_grad_()
+
+        def fn(x):
+            return torch.sparse.softmax(x, 0).to_dense()
```

Comment:
Can we use straight gradcheck now that https://github.com/pytorch/pytorch/pull/95405 merged?

Reply:
Perhaps even enable OpInfo coverage?

Reply:
Gradcheck still does not support sparse outputs, so this wrapper is still needed.

Reply:
@pearu - Can we extend gradcheck to work for this case too?

Reply:
This works:
```python
>>> torch.autograd.gradcheck(lambda x: torch.sparse.softmax(x, 0).to_dense(), t, masked=True, check_sparse_nnz=True)
True
```

Here's a list of tasks for extending gradcheck:
1. Eliminate `check_sparse_nnz=` kw argument as redundant. This is an easy fix.
2. Allow op to return sparse output. I tried to disable the check --- there appear serious issues that need to be resolved to get sparse output working.

Reply:
Also nit: can we enable `float` dtype as well?

Reply:
@nikitaved float32 is typically disabled for gradcheck tests because the numerical Jacobian result is often too inaccurate. To make the test work, tweaking of default eps/atol/rtol parameters is required.

Reply:
These are trivial cases with no computation, so it should work for any dtype I reckon.

Reply:
Also curious if the "real" sparse softmax tests need to do all the stuff they are doing with computing and comparing the full Jacobian. Can't we just use gradcheck there as well? If we simplify that test then this just becomes an additional shape + nnz to pass as a test case.

Reply:
I suggest keeping "the real sparse softmax tests" because it provides a closed form jacobian (as defined in `softmax_jacobian_analytic`) that would be the thrid independent approach of testing "sparse softmax" gradients together with the numerical and analytical jacobians from gradcheck.